src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,299 @@
+    the retrieved data.
+    Checks SAS UUT responds with HeartbeatResponse either GRANT_TERMINATED or INVALID_VALUE,
+    indicating a terminated Grant.
+
+    """
+
+    config = loadConfig(config_filename)
+
+    device_c2 = config['registrationRequestC2']
+    device_c4 = config['registrationRequestC4']
+    grant_g2 = config['grantRequestG2']
+    grant_g4 = config['grantRequestG4']
+    sas_test_harness_dump_records = [config['sasTestHarnessConfigDumpRecords']['cbsdRecords'],
+                                     config['sasTestHarnessConfigDumpRecords']['ppaRecords'],
+                                     config['sasTestHarnessConfigDumpRecords']['escSensorRecords']]
+
+    # Initialize SAS Test Harness Server instance to dump FAD records
+    sas_test_harness = SasTestHarnessServer(config['sasTestHarnessConfig']['sasTestHarnessName'],
+                                            config['sasTestHarnessConfig']['hostName'],
+                                            config['sasTestHarnessConfig']['port'],
+                                            config['sasTestHarnessConfig']['serverCert'],
+                                            config['sasTestHarnessConfig']['serverKey'],
+                                            config['sasTestHarnessConfig']['caCert'])
+    
+    sas_test_harness.writeFadRecords(sas_test_harness_dump_records)
+
+    # Start the server
+    sas_test_harness.start()
+
+    # Whitelist the FCCID and UserID for CBSD C2 in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_c2['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_c2['userId']})
+
+    # Whitelist the FCCID and UserID for CBSD C4 in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_c4['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_c4['userId']})
+
+    # Whitelist the FCCID and UserID of the devices loaded to SAS-test_harness in SAS UUT
+    for cbsdRecord in config['sasTestHarnessConfigDumpRecords']['cbsdRecords']:
+      self._sas_admin.InjectFccId({'fccId': cbsdRecord['registration']['fccId']})
+
+    
+    # Pre-load conditional registration data for C2 and C4 CBSDs.
+    if ('conditionalRegistrationData' in config) and (
+        config['conditionalRegistrationData']):
+      self._sas_admin.PreloadRegistrationData(
+          config['conditionalRegistrationData'])
+
+    # Register devices C2 and C4, request grants G2 and G4 respectively with SAS UUT.
+    # Ensure the registration and grant requests are successful.
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted([device_c2, device_c4],
+                                                          [grant_g2, grant_g4])
+
+    # Send the Heartbeat request for the Grant G2 and G4 of CBSD C2 and C4
+    # respectively to SAS UUT.
+    transmit_expire_times = self.assertHeartbeatsSuccessful(cbsd_ids, grant_ids,
+                                      [[('GRANTED')],[('GRANTED')]])
```

Comment:
Shouldn't it just be `['GRANTED', 'GRANTED']`? Please check carefully against the function definition.